### PR TITLE
Manage Tags Bugs

### PR DESCRIPTION
**Pull request by ahsu1230**
Few things
- clicking on tag now goes to user's search tag page (with tag name)
- when selecting a library to add keeps to, modal pops up (stay on manage page or go to library)
- refactor manage tags state out of factory & into controller
- fixing bug for adding tagged Keep to Library => if a Keep with URI already exists in a library but is inactive, then adding keep to library will cause server error (duplicate URIs). So remember to reactivate keep isntead of adding a new one.
- small edit to query for sorting tags based on number of keeps

Sorry so many things on one PR. I should try to fix one bug at a time =P
@andrewconner @yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/CollectionRepo.scala [hunk 1]
```diff
@@ -173,7 +173,7 @@ class CollectionRepoImpl @Inject() (
   def getByUserSortedByNumKeeps(userId: Id[User], page: Int, size: Int)(implicit session: RSession): Seq[(CollectionSummary, Int)] = {
     import StaticQuery.interpolation
     import scala.collection.JavaConversions._
-    val query = sql"select c.id, c.external_id, c.name, count(kc.id) as keep_count from collection c, keep_to_collection kc where kc.collection_id = c.id and user_id=${userId} and kc.state = ${KeepToCollectionStates.ACTIVE} and c.state=${CollectionStates.ACTIVE} group by c.id order by keep_count desc limit ${size} offset ${page * size}"
+    val query = sql"select c.id, c.external_id, c.name, count(kc.id) as keep_count from collection c, keep_to_collection kc, bookmark b where kc.collection_id=c.id and kc.bookmark_id=b.id and c.user_id=${userId} and b.state=${KeepStates.ACTIVE} and kc.state=${KeepToCollectionStates.ACTIVE} and c.state=${CollectionStates.ACTIVE} group by c.id order by keep_count desc limit ${size} offset ${page * size}"
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/model/CollectionRepo.scala:176]: thanks for the help! @raymondng

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagToLib.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+.directive('kfTagLibrary', ['$location',
+  function ($location) {
+    return {
+      restrict: 'A',
+      require: '^kfModal',
+      templateUrl: 'tagManage/tagToLibMsg.tpl.html',
+      link: function (scope, element, attrs, kfModalCtrl) {
+
+        scope.hide = function () {
+          kfModalCtrl.close();
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagToLib.js:14]: Come talk to me about this. I think that there may be a cleaner way of closing this modal.

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagToLibMsg.tpl.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div class="dialog-content">
+  <a class="dialog-x sprite-close-x-icon" ng-click="hide()"></a>
+  <div class="dialog-body-wrap">
+    <div class="dialog-body dialog-centered">
+        <div>We're copying your tagged keeps over, but it might take a second...</div>
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagToLibMsg.tpl.html]: Fix indentation on this and the next line.

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js [hunk 4]
```diff
@@ -120,5 +143,9 @@ angular.module('kifi')
       }
     };
 
+    $scope.navigateToTag = function (tagName) {
+      $location.path('/find').search('q','tag:' + tagName);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js]: Add space after comma.

**Comment by yipingliao** [comment]
Nice! I am looking forward to seeing this in action. :)

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagToLibModal.tpl.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<div kf-modal class="kf-tag-lib-modal">
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagToLibModal.tpl.html:1]: Add `force-close="forceClose"`

File: kifi-backend/modules/shoebox/app/com/keepit/model/CollectionRepo.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/model/CollectionRepo.scala:176]: Thanks Ray

**Pull request merged**